src/overlays/actors/ovl_Obj_Boyo/z_obj_boyo.c

```diff
@@ -19,22 +26,183 @@ const ActorInit Obj_Boyo_InitVars = {
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x01CBFFBE, 0x00, 0x00 },
+        TOUCH_NONE,
+        BUMP_ON,
+        OCELEM_ON,
+    },
+    {
+        60,
+        140,
+        0,
+        { 0, 0, 0 },
+    },
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 300, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 300, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+static ObjBoyoUnkStruct gCollisionHandling[] = {
+    { ACTOR_PLAYER, ObjBoyo_PushPlayer },
+    { ACTOR_EN_KAIZOKU, ObjBoyo_PushPirate },
+    { ACTOR_EN_BOM, ObjBoyo_ExplodeBomb },
+};
+
+void ObjBoyo_Init(Actor* thisx, PlayState* play) {
+    ObjBoyo* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    Collider_InitCylinder(play, &this->collider);
+    Collider_SetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    this->actor.colChkInfo.mass = 0xFF;
+    this->animatedMaterial = Lib_SegmentedToVirtual(&object_boyo_Matanimheader_000E88);
+}
+
+void ObjBoyo_Destroy(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    ObjBoyo* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+void ObjBoyo_PushPlayer(ObjBoyo* this, Actor* actor) {
+    ((Player*)actor)->pushedSpeed = 30.0f;
+    ((Player*)actor)->pushedYaw = (s16)this->actor.yawTowardsPlayer;
+}
+
+void ObjBoyo_PushPirate(ObjBoyo* this, Actor* actor) {
+    EnKaizoku* kaizoku = (EnKaizoku*)actor;
+
+    kaizoku->unk_2F0 = 30.0f;
+    kaizoku->unk_2F4 = Actor_WorldYawTowardActor(&this->actor, &kaizoku->picto.actor);
+}
+
+void ObjBoyo_ExplodeBomb(ObjBoyo* this, Actor* actor) {
+    EnBom* bom = (EnBom*)actor;
+
+    bom->timer = 0;
+}
+
+/*
+ * Find the Actor we collided with.
+ */
+Actor* ObjBoyo_FindCollidedActor(ObjBoyo* this, PlayState* play, s32* index) {
+    Actor* collidedActor;
+    ObjBoyoUnkStruct* data;
+    s32 counter;
+
+    if (this->collider.base.ocFlags2 & OC2_HIT_PLAYER) {
+        *index = 0;
+        // Returns first actor of the actor List for actors of type Player.
+        // In short it gets the player's Actor*.
+        return (s32)play->actorCtx.actorLists[2].first;
+    }
+    data = gCollisionHandling + 1;
+    if (this->collider.base.ocFlags1 & OC1_HIT) {
+        collidedActor = this->collider.base.oc;
+        for (counter = 1; counter != 3; ++counter) {
+            if (collidedActor->id == (*data).id) {
+                *index = counter;
+                return collidedActor;
+            }
+            data += 1;
+        }
+    }
+    return 0;
+}
+
+void ObjBoyo_Update(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    ObjBoyo* this = THIS;
+    Actor* collidedActor;
+    s32 index;
+
+    collidedActor = ObjBoyo_FindCollidedActor(this, play, &index);
+
+    if (collidedActor != 0) {
```

Comment:
```suggestion
    if (collidedActor != NULL) {
```
Since `collidedActor` is `NULL`